src/network/clientpackethandler.cpp

```diff
@@ -1409,6 +1409,23 @@ void Client::handleCommand_OverrideDayNightRatio(NetworkPacket* pkt)
 	m_client_event_queue.push(event);
 }
 
+void Client::handleCommand_LocalPlayerSprite(NetworkPacket* pkt)
+{
+	LocalPlayer *player = m_env.getLocalPlayer();
+	assert(player != NULL);
+
+	*pkt >> player->local_sprite_tx_basepos[0];
+	*pkt >> player->local_sprite_tx_basepos[1];
+	*pkt >> player->local_sprite_tx_basepos[2];
+	*pkt >> player->local_sprite_tx_basepos[3];
+	*pkt >> player->local_sprite_num_frames[0];
+	*pkt >> player->local_sprite_num_frames[1];
+	*pkt >> player->local_sprite_num_frames[2];
+	*pkt >> player->local_sprite_num_frames[3];
+	*pkt >> player->local_sprite_framelength;
+	*pkt >> player->local_sprite_select_horiz_by_yawpitch;
```

Comment:
shouldn't all this be an active object message?

Reply:
I just copied it from set_local_animation, not really understanding the difference between a network packet send and a gob_cmd (AOM) like it is done with set_sprite_framelength.

But that would make sense and solve a small issue I set aside: when connecting the player's animation is set to NO_ANIM, then when set_local_sprite_animation is called it is not updated because the animation hasn't changed. So select_horiz_by_yawpitch may be ignored until the animation changes (set_sprite would force-update this but it may differ). As a result the player is facing the camera when switching to 3rd person view before moving. And I refused to put some code in the handleCommand.

If there is no real reason set_local_animation is a raw networkpacket rather than an AOM, I'll update this. And maybe start an other branch to switch set_local_animation as well.

Reply:
Yeah, I don't see a reason why this needs to be a network packet.

Also, have you tested this in multiplayer? How would the sprite parameters for a non-local player be set?

Reply:
Ok, let's do this.

In multiplayer the local sprite overwrites the player's sprite. For non local players the function set_sprite was already there, but only for LuaEntities. With these patches players can also accept set_sprite. Without them you can set {visual = "sprite"} in the player's properties but not assign an animation to them.

I checked by running 2 instances of Minetest locally, one hosting the server and the other joining localhost with different parameters to set_sprite and set_local_sprite and see if the players are not rendered with the same animation for the current player and the other player.

Reply:
Looking deeper at AOM I think I've understood why set_local_anim uses a network packet directly.

If I read correctly, AOM are bound to an ObjectRef, and are sent to all clients that cares about it. But set_local_* is targeted only to the RemotePlayer's client. Using an AOM would spam other clients or there should be a filter to send it only to the targeted client.

Reply:
Yeah that sounds correct. You can keep it like this then.